### PR TITLE
Prepare SQL tests for running them from Jet

### PR DESCRIPTION
**Pull request by viliam-durina**

### CODE REVIEW

**Review by taburet — APPROVED** [review]

File: hazelcast-sql/src/test/java/com/hazelcast/sql/impl/plan/cache/PlanCacheIntegrationTest.java [hunk 1]
```diff
@@ -45,7 +45,7 @@
 @Category({QuickTest.class, ParallelJVMTest.class})
 public class PlanCacheIntegrationTest extends PlanCacheTestSupport {
 
-    private final TestHazelcastInstanceFactory factory = new TestHazelcastInstanceFactory(2);
+    private final SqlTestInstanceFactory factory = SqlTestInstanceFactory.create();
```
Comment (devozerov) [inline hazelcast-sql/src/test/java/com/hazelcast/sql/impl/plan/cache/PlanCacheIntegrationTest.java:48]: Plan cache tests are not related to expressions. Do you really want to run it in Jet environment?

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2070/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2070/Hazelcast-pr-builder-2070.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.1:checkstyle (default) on project hazelcast-sql: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There is 1 error reported by Checkstyle 8.29 with /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder_3/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder_3/hazelcast-sql/src/test/java/com/hazelcast/sql/SqlTestInstanceFactory.java:77:9: Redundant 'public' modifier. [RedundantModifier]
--------------------------
</details>
</pre>

**Comment by hz-highfive** [comment]
The job `Hazelcast-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2072/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/2072/Hazelcast-pr-builder-2072.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.1:checkstyle (default) on project hazelcast-sql: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There is 1 error reported by Checkstyle 8.29 with /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-builder/hazelcast-sql/src/test/java/com/hazelcast/sql/SqlTestInstanceFactory.java:77:9: Redundant 'public' modifier. [RedundantModifier]
--------------------------
</details>
</pre>

**Comment by devozerov** [comment]
Another possible solution, that is definitely reliable - is to run all SQL tests under Jet environment. This will require no manual intervention, no naming conventions, and will not depend on how the developer names the test. 
Currently, all SQL tests take about 6 mins to run. And of these 6 minutes, 4 minute is dedicated to a single SQL index test, that I will fix soon.

**Comment by viliam-durina** [comment]
I've included all the tests. Measured on my local machine, the selected tests take 2:43, all tests without the SqlIndexTest take 3:19, so it's acceptable. Maybe we'll also run the imdg tests only in nightly build in Jet.

File: hazelcast-sql/src/test/java/com/hazelcast/sql/impl/plan/cache/PlanCacheIntegrationTest.java (hunk 1 above)
Comment (viliam-durina) [reply hazelcast-sql/src/test/java/com/hazelcast/sql/impl/plan/cache/PlanCacheIntegrationTest.java:48]: It should not hurt...

Comment (viliam-durina) [reply hazelcast-sql/src/test/java/com/hazelcast/sql/impl/plan/cache/PlanCacheIntegrationTest.java:48]: Running the tests is not about expressions, but about jet modifying the optimization.

**Review by devozerov — APPROVED** [review]

**Pull request merged**